src/main/scala/circt/stage/phases/CIRCT.scala

```diff
@@ -126,34 +135,63 @@ class CIRCT extends Phase {
-          case Some(CIRCTTarget.Verilog)       => None
-          case Some(CIRCTTarget.SystemVerilog) => None
-          case None =>
+        ((circtOptions.target, split) match {
+          case (Some(CIRCTTarget.FIRRTL), false)        => Seq("-ir-fir")
+          case (Some(CIRCTTarget.HW), false)            => Seq("-ir-hw")
+          case (Some(CIRCTTarget.Verilog), true)        => Seq("--split-verilog", s"-o=${stageOptions.targetDir}")
+          case (Some(CIRCTTarget.Verilog), false)       => None
+          case (Some(CIRCTTarget.SystemVerilog), true)  => Seq("--split-verilog", s"-o=${stageOptions.targetDir}")
+          case (Some(CIRCTTarget.SystemVerilog), false) => None
+          case (None, _) =>
             throw new Exception(
               "No 'circtOptions.target' specified. This should be impossible if dependencies are satisfied!"
             )
+          case (_, true) =>
+            throw new Exception(
+              s"The circtOptions.target specified (${circtOptions.target}) does not support running with an EmitAllModulesAnnotation as CIRCT only supports one-file-per-module for Verilog or SystemVerilog targets."
+            )
+          case _ =>
+            throw new Exception(
+              s"Invalid combination of circtOptions.target ${circtOptions.target} and split ${split}"
+            )
         })
 
     try {
       logger.info(s"""Running CIRCT: '${cmd.mkString(" ")} < $$input'""")
-      val result = (cmd #< new java.io.ByteArrayInputStream(input.getBytes)).!!
-
-      circtOptions.target match {
-        case Some(CIRCTTarget.FIRRTL) =>
-          Seq(EmittedMLIR(outputFileName, result, Some(".fir.mlir")))
-        case Some(CIRCTTarget.HW) =>
-          Seq(EmittedMLIR(outputFileName, result, Some(".hw.mlir")))
-        case Some(CIRCTTarget.Verilog) =>
-          Seq(EmittedVerilogCircuitAnnotation(EmittedVerilogCircuit(outputFileName, result, ".v")))
-        case Some(CIRCTTarget.SystemVerilog) =>
-          Seq(EmittedVerilogCircuitAnnotation(EmittedVerilogCircuit(outputFileName, result, ".sv")))
-        case None =>
-          throw new Exception(
-            "No 'circtOptions.target' specified. This should be impossible if dependencies are satisfied!"
-          )
+      println(s"""Running CIRCT: '${cmd.mkString(" ")} < $$input'""")
+      val stdoutStream, stderrStream = new java.io.ByteArrayOutputStream
+      val stdoutWriter = new java.io.PrintWriter(stdoutStream)
+      val stderrWriter = new java.io.PrintWriter(stderrStream)
+      val exitValue = (cmd #< new java.io.ByteArrayInputStream(input.getBytes))
+        .!(ProcessLogger(stdoutWriter.println, stderrWriter.println))
+      stdoutWriter.close()
+      stderrWriter.close()
+      val result = stdoutStream.toString
+      logger.info(result)
+      val errors = stderrStream.toString
+      if (exitValue != 0) {
+        StageUtils.dramaticError(s"${binary} failed.\nExitCode:\n${exitValue}\nSTDOUT:\n${result}\nSTDERR:\n${errors}")
```

Comment:
copying @seldridge comment from #14 : 

It might be cleaner to change this to throw a java.io.IOException. In the catch below, the current java.io.IOException should probably be a java.io.FileNotFoundException, but I'd have to see what actually gets thrown. This would then prevent doing error handling code in the try portion. This would require moving the streams outside the try so they are available to both.

logger.error is probably better, but the current code doesn't use this. I propose using StageError for now and changing it to logger.error in a subsequent patch.